src/main/kotlin/com/sourcegraph/cody/chat/ui/MessagePart.kt

```diff
@@ -38,20 +38,23 @@ class CodeEditorPart(
     get() = _text.get()
 
   fun updateCode(project: Project, code: String, language: String?) {
-    updateLanguage(language)
+    recognizeLanguage(language)
     updateText(project, code)
   }
 
-  fun updateLanguage(language: String?) {
-    val fileType: FileType =
-        Language.getRegisteredLanguages()
-            .firstOrNull { it.displayName.equals(language, ignoreCase = true) }
-            ?.let { FileTypeManager.getInstance().findFileTypeByLanguage(it) }
-            ?: PlainTextFileType.INSTANCE
-    val editorHighlighter =
-        HighlighterFactory.createHighlighter(
-            fileType, EditorColorsManager.getInstance().schemeForCurrentUITheme, null)
-    editor.highlighter = editorHighlighter
+  fun recognizeLanguage(languageName: String?) {
+    if (recognizedLanguage != null) return
+    val language = Language.getRegisteredLanguages()
+          .filter { it != Language.ANY }
```

Comment:
Note: This `Language.ANY` filter is required due to false-positive that may randomly occur on blank/empty language name.